src/App.jsx

```diff
@@ -13,46 +15,90 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_ALHPABET = 'alpha';
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const getSortedGoods = (listOfGoods, { sortValue, reverseValue }) => {
+  const copyOfGoods = [...listOfGoods];
+
+  if (sortValue) {
+    copyOfGoods.sort((goodA, goodB) => {
+      switch (sortValue) {
+        case SORT_BY_ALHPABET:
+          return goodA.localeCompare(goodB);
+        case SORT_BY_LENGTH:
+          return goodA.length - goodB.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseValue) {
+    copyOfGoods.reverse();
+  }
+
+  return copyOfGoods;
+};
+
+export const App = () => {
+  const [sortValue, setsortValue] = useState('');
+  const [isReversed, setIsReversed] = useState('');
+  const goodsForRender = getSortedGoods(goodsFromServer,
+    { sortValue, reverseValue: isReversed });
+
+  const resetFilters = () => {
+    setsortValue('');
+    setIsReversed('');
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', { 'is-light':
+          sortValue !== SORT_BY_ALHPABET })}
+          onClick={() => setsortValue(SORT_BY_ALHPABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', { 'is-light':
+          sortValue !== SORT_BY_LENGTH })}
+          onClick={() => setsortValue(SORT_BY_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', { 'is-light': !isReversed })}
+          onClick={() => (isReversed
+            ? setIsReversed('')
+            : setIsReversed(!isReversed))}
```

Comment:
```suggestion
          onClick={() => setIsReversed((prev) => !prev))}
```
If you need to toggle value use functional setState